hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java

```diff
@@ -117,12 +121,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       acquiredLock = omMetadataManager.getLock().acquireLock(BUCKET_LOCK,
           volumeName, bucketName);
 
-      // Not doing bucket/volume checks here. In this way we can avoid db
-      // checks for them.
-      // TODO: Once we have volume/bucket full cache, we can add
-      // them back, as these checks will be inexpensive at that time.
-      OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable().get(objectKey);
+      // Check volume exist.
+      if (omMetadataManager.getVolumeTable().isExist(volumeName)) {
```

Comment:
Here it should be if (!omMetadataManager.getVolumeTable().isExist(volumeName)) right?

As here if it does not exist, we should return error?

```
  /**
   * Check if a given key exists in Metadata store.
   * (Optimization to save on data deserialization)
   * A lock on the key / bucket needs to be acquired before invoking this API.
   * @param key metadata key
   * @return true if the metadata store contains a key.
   * @throws IOException on Failure
   */
  boolean isExist(KEY key) throws IOException;
```

Reply:
And also we can do little optimization here, check the first bucket exists or not, if it does not exist, then check volume?

Reply:
Thanks @bharatviswa504 
I update the PR, and also add some test aiming to the change.